v-next/hardhat/src/internal/builtin-plugins/network-manager/hook-handlers/config.ts

```diff
@@ -60,35 +71,82 @@ export async function resolveUserConfig(
-      type: "http",
-      chainId: networkConfig.chainId,
-      chainType: networkConfig.chainType,
-      from: networkConfig.from,
-      gas: resolveGasConfig(networkConfig.gas),
-      gasMultiplier: networkConfig.gasMultiplier ?? 1,
-      gasPrice: resolveGasConfig(networkConfig.gasPrice),
-      url: networkConfig.url,
-      timeout: networkConfig.timeout ?? 20_000,
-      httpHeaders: networkConfig.httpHeaders ?? {},
-    };
-
-    resolvedNetworks[networkName] = resolvedNetworkConfig;
+    if (networkConfig.type === "http") {
+      const resolvedNetworkConfig: HttpNetworkConfig = {
+        type: "http",
+        chainId: networkConfig.chainId,
+        chainType: networkConfig.chainType,
+        from: networkConfig.from,
+        gas: resolveGasConfig(networkConfig.gas),
+        gasMultiplier: networkConfig.gasMultiplier ?? 1,
+        gasPrice: resolveGasConfig(networkConfig.gasPrice),
+        url: networkConfig.url,
+        timeout: networkConfig.timeout ?? 20_000,
+        httpHeaders: networkConfig.httpHeaders ?? {},
+      };
+
+      resolvedNetworks[networkName] = resolvedNetworkConfig;
+    }
+
+    if (networkConfig.type === "edr") {
+      const resolvedNetworkConfig: EdrNetworkConfig = {
+        type: "edr",
+        chainId: networkConfig.chainId,
+        chainType: networkConfig.chainType ?? "l1",
+        from: resolveFrom(networkConfig.from),
+        gas: resolveGasConfig(networkConfig.gas),
+        gasMultiplier: networkConfig.gasMultiplier ?? 1,
+        gasPrice: resolveGasConfig(networkConfig.gasPrice),
+
+        hardfork: networkConfig.hardfork ?? "cancun",
+        networkId: networkConfig.networkId ?? networkConfig.chainId,
+        blockGasLimit: networkConfig.blockGasLimit ?? 12_500_000,
+        minGasPrice: BigInt(networkConfig.minGasPrice ?? 0),
+        automine: networkConfig.automine ?? true,
+        intervalMining: networkConfig.intervalMining ?? 0,
+        mempoolOrder: networkConfig.mempoolOrder ?? "fifo",
+        chains: networkConfig.chains ?? new Map(),
+        genesisAccounts: networkConfig.genesisAccounts ?? [],
+        allowUnlimitedContractSize:
+          networkConfig.allowUnlimitedContractSize ?? false,
+        throwOnTransactionFailures:
+          networkConfig.throwOnTransactionFailures ?? true,
+        throwOnCallFailures: networkConfig.throwOnCallFailures ?? true,
+        allowBlocksWithSameTimestamp:
+          networkConfig.allowBlocksWithSameTimestamp ?? false,
+        enableTransientStorage: networkConfig.enableTransientStorage ?? false,
+        enableRip7212: networkConfig.enableRip7212 ?? false,
+      };
+
+      resolvedNetworks[networkName] = resolvedNetworkConfig;
+    }
   }
 
   return {
     ...resolvedConfig,
     defaultChainType: resolvedConfig.defaultChainType ?? "unknown",
-    defaultNetwork: resolvedConfig.defaultNetwork ?? "localhost",
+    defaultNetwork: resolvedConfig.defaultNetwork ?? "hardhat",
```

Comment:
Switch the default network type to `hardhat` to match Hardhat v2, should this also imply a change to the default chain type.